pipeline_steps/deploy.groovy

```diff
@@ -65,6 +65,26 @@ def upgrade(Map args) {
   ) // conditionalStage
 }
 
+def upgrade_leap(Map args) {
```

Comment:
This creates a lot of code duplication, what about something like:
```groovy
void upgrade(String stage_name, String upgrade_script, List env_vars) {
  common.conditionalStage(
    stage_name: stage_name,
    stage: {
      environment_vars = env_vars + common.get_deploy_script_env()
      withEnv(environment_vars){
        dir("/opt/rpc-openstack/openstack-ansible"){
          sh "git reset --hard"
        }   
        common.prepareRpcGit()
        dir("/opt/rpc-openstack"){
          sh """ 
            scripts/$upgrade_script
          """
        } // dir
      } // withEnv
    } // stage
  ) // conditionalStage
}

void upgrade_major(Map args) {
  upgrade("Major Upgrade", "test-upgrade.sh", args.environment_vars)
}

void upgrade_leapfrog(Map args) {
  upgrade("Leapfrog Upgrade", "leapfrog/ubuntu14-leapfrog.sh", args.environment_vars)
}
```

If the code needs to diverge in the future, we can always look to separate the implementations. For now I think this will be easier to manage.